logging/tests/unit/test__helpers.py

```diff
@@ -105,15 +105,23 @@ def test_metadata_does_not_exist(self):
         self.assertIsNone(metadata)
 
     def test_request_exception(self):
+        import requests
+
         metadata_key = 'test_url_cannot_connect'
         metadata_url = 'http://metadata.invalid/'
 
-        patch = mock.patch(
+        requests_get_mock = mock.Mock()
```

Comment:
Can you set the `spec`?

Reply:
It is being used to mock a single function:  do we set `spec=['__call__']` as a rule in such cases?  I'd originally written it as an inline function:

```python
        def requests_get(*args, **kw):
              raise requests.exceptions.RequestsException('test')
```

which seemed clearer, but you have an allergy for them. :)

Reply:
`git grep` showed a number of such cases, so I followed suit in 52c4b9f.